src/main/scala/org/graphframes/lib/ConnectedComponents.scala

```diff
@@ -313,12 +313,18 @@ object ConnectedComponents extends Logging {
 
     var converged = false
     var iteration = 1
-    var prevSum: BigDecimal = null
+
+    // compute min neighbors (including self-min)
+    var minNbrs1: DataFrame = minNbrs(ee) // src >= min_nbr
```

Comment:
This makes sense to do the first minimization vector connection reduction outside of the while loop. +1